pkg/kubelet/kuberuntime/kuberuntime_container.go

```diff
@@ -474,6 +474,43 @@ func (m *kubeGenericRuntimeManager) executePreStopHook(pod *v1.Pod, containerID
 	return int64(metav1.Now().Sub(start.Time).Seconds())
 }
 
+// executeRetryUntilSuccess runs the Retry-Until lifecycle hooks if applicable and returns the duration it takes.
+func (m *kubeGenericRuntimeManager) executeRetryUntilSuccessHook(pod *v1.Pod, containerID kubecontainer.ContainerID, containerSpec *v1.Container, gracePeriod int64) int64 {
```

Comment:
@timstclair in executePreStopHook func gracePeriod is int64 type,and in all killcontainer func graceperiod initialized to init64, so if i change executeRetryUntilSuccessHook return type, means I need to modify all the values used in the killcontainer function to graceperiod. If this is what you need, I can modify.